pcbnew2boardview.py

```diff
@@ -139,17 +139,44 @@ def convert(pcb, brd):
     brd.write("\n")
 
 
+def convert_obdata(pcb, obdata):
+    obdata.write("COMPONENTS_DATA_START\n")
+    obdata.write("### Component Category Value Comment\n")
+    obdata.write("### v = value, p = package, c = manufacturer code, r = rating, m = misc, s = status\n")
+    obdata.write("###\n")
+    
+    for module in pcb.GetFootprints():
+        ref = module.GetReference()
+        package = module.GetFPIDAsString()
+        value = module.GetValue()
+        
+        package = re.sub(r'^.*:', '', package)
+            
+        obdata.write(f"{ref} p {package}\n")
+        if value == "NO-FIT" or value == "DNP":
+            obdata.write(f"{ref} s -\n")
+        obdata.write(f"{ref} v {value}\n")
+        
+    obdata.write("COMPONENTS_DATA_END\n")
+    obdata.write("### END")
+
+
 def main():
     parser = argparse.ArgumentParser()
     parser.add_argument(
         "kicad_pcb_file", metavar="KICAD-PCB-FILE", type=str,
         help="input in .kicad_pcb format")
     parser.add_argument(
-        "brd_file", metavar="BRD-FILE", type=argparse.FileType("wt"),
+        "brd_file", metavar="BRD-FILE", type=argparse.FileType("w"),
         help="output in .brd format")
-
+    parser.add_argument(
+        "--obdata", metavar="OBDATA-FILE", type=argparse.FileType("w"),
+        help="output in .obdata format")
+    
     args = parser.parse_args()
-    convert(pcbnew.LoadBoard(args.kicad_pcb_file), args.brd_file)
+    board = pcbnew.LoadBoard(args.kicad_pcb_file)
+    convert_brd(board, args.brd_file)
+    convert_obdata(board, args.obdata)
```

Comment:
I think this will fail if `--obdata` is not provided. I think you'll need to figure out how to use the script with KiCad (I don't know how to do it on Windows).

Reply:
I put the convert_obdata command within an if statement and set its default value to none, though I'm not sure if this fixes what you're talking about.

Reply:
So `default=None` is the default, but yeah this works.

Reply:
Gotcha so I'll omit that and keep the if statement just to avoid redundancy.